approaches/finetuning_tasks_test.py

```diff
@@ -0,0 +1,87 @@
+import os
+from typing import List
+
+import numpy as np
+
+from approaches import AbstractApproach
+from text_models import AbstractModel, BertSiameseModel
+
+
+class FinetuningTasksTest(AbstractApproach):
+    def __init__(
+        self,
+        approach: AbstractApproach,
+        tasks: List[str],
+        models_directory: str,
+        model_suffix: str,
+    ):
+        super().__init__(approach.train, approach.test)
+        self.tasks = tasks
+        self.models_directory = models_directory
+        self.model_suffix = model_suffix
+        self.approach = approach
+
+        self.all_results = None
+        self.results_list = None
+
+    def evaluate_all(
+        self,
+        from_scratch_model: AbstractModel,
+        pretrained_model: AbstractModel,
+        doc_task_model: AbstractModel,
+        finetuned_model: AbstractModel,
+        topns: List[int],
+        silence: bool = False,
+    ):
+        self.approach.evaluate_all(
+            from_scratch_model, pretrained_model, doc_task_model, finetuned_model, topns, silence=True
+        )
+        self.results_list = [self.approach.results]
+
+        if finetuned_model is None and doc_task_model is None:
+            return
+
+        self.all_results = self.approach.results.copy()
+        self.all_results.rename(columns={AbstractModel.finetuned: f"PT+DOC({self.tasks[0]})+TASK"}, inplace=True)
+
+        for i in range(1, len(self.tasks)):
```

Comment:
We need to skip first task because we have already processed this task with the self.approach.evaluate_all() method call above. I have added comment about this to the code